oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -326,6 +328,32 @@ public void getDefaultCredentials_envUser_providesToken() throws IOException {
     testUserProvidesToken(testProvider, USER_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN);
   }
 
+  @Test
+  public void getDefaultCredentials_quota_project() throws IOException {
+    InputStream userStream =
+        UserCredentialsTest.writeUserStream(
+            USER_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN, QUOTA_PROJECT);
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    String userPath = tempFilePath("user.json");
+    testProvider.addFile(userPath, userStream);
+    testProvider.setEnv(DefaultCredentialsProvider.CREDENTIAL_ENV_VAR, userPath);
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.addClient(USER_CLIENT_ID, USER_CLIENT_SECRET);
+    transportFactory.transport.addRefreshToken(REFRESH_TOKEN, ACCESS_TOKEN);
+
```

Comment:
get credential before env is set and check quota is not present or different

Reply:
yeah wrote the test exactly like that first. Then realized that there is a cache. So we will get back same cred even if env var is set.

Reply:
What cache? Then this test is not effective?

Reply:
Cache is [here](https://github.com/googleapis/google-auth-library-java/blob/main/oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java#L119). This test calls getDefaultCredentials only once. The second check is just seeing if user can update (new copy) the quota project on the credential to get the value they want.